wapitiCore/attack/mod_methods.py

```diff
@@ -64,16 +106,44 @@ async def attack(self, request: Request, response: Optional[Response] = None):
         if response.is_success or response.is_redirect:
             methods = response.headers.get("allow", '').upper().split(',')
             methods = {method.strip() for method in methods if method.strip()}
-            interesting_methods = sorted(methods - self.KNOWN_METHODS)
+            interesting_methods, methods_responses = await self.test_compare_methods(request,
+                                                                                     methods - self.KNOWN_METHODS)
 
             if interesting_methods:
                 log_orange("---")
-                log_orange(f"Interesting methods allowed on {page}: {', '.join(interesting_methods)}")
+                # methods returned by the OPTIONS method request
+                option_log = f"Interesting methods allowed on {page}: {', '.join(interesting_methods)}"
+                log_orange(option_log)
                 await self.add_addition(
                     category=NAME,
                     request=option_request,
-                    info=f"Interesting methods allowed on {page}: {', '.join(interesting_methods)}",
+                    info=option_log,
                     wstg=WSTG_CODE,
                     response=response
                 )
+                # if a method has a relevant information, it will be listed
+                for method, data in methods_responses.items():
+                    if not data['status_different'] and not data['content_different']:
+                        continue
+                    logging_string = f"Method {method} returned "
+                    differences_str = []
+                    if data['status_different']:
+                        differences_str.append(f"{data['code']} server code")
+                    if data['content_different']:
+                        differences_str.append("a body content")
+                    logging_string += f"{' and '.join(differences_str)} different from GET method on {page}"
+                    if logging_string:
+                        log_orange(logging_string)
+                        await self.add_addition(
+                            category=NAME,
+                            request=Request(
+                                page,
+                                method,
+                                referer=request.referer,
+                                link_depth=request.link_depth
+                            ),
+                            info=logging_string,
+                            wstg=WSTG_CODE,
+                            response=data['resp']
+                        )
```

Comment:
You made this part of method way smarter than before, we should refactor this before before you go right into an indentation hell. 

What I recommend is transforming the structure : 

```python
if A
   if B
     if C
```

into : 
```python
if not A: return
if not B: return
if not C: return 
```